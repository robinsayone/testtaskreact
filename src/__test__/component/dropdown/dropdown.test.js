import React from 'react';
import ReactDom from 'react-dom';
import { render,cleanup , screen ,fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';

import BasicSelect from './../../../component/dropdown/dropdown';

afterEach(cleanup);

it("dropdownrender without crashing",()=>{
    const div = document.createElement("div");
    ReactDom.render(<BasicSelect></BasicSelect>,div);
});
it("dropdown renders correctly" , () =>{
    render(<BasicSelect label={"dropdown"}></BasicSelect>);
    expect(screen.getByText(/dropdown/i)).toBeInTheDocument();

});

it("dropdown onchange works correctly",()=>{
    const samefun = jest.fn();
    render(<BasicSelect label={"dropdown"} onChange={samefun}></BasicSelect>);
    const element = screen.getByTestId("dropdowntestId").querySelector('input');
    fireEvent.change(element,{target:{value:'Health213'}});          

});

it("dropdown matches snapshot",()=>{
    const tree = renderer.create(<BasicSelect></BasicSelect>).toJSON();
    expect(tree).toMatchSnapshot();
});