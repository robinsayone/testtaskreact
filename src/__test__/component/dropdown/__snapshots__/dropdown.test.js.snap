// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown matches snapshot 1`] = `
<div
  className="MuiBox-root css-ece9u5"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      id="demo-simple-select-label"
    />
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      data-testid="dropdowntestId"
      onClick={[Function]}
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-label demo-simple-select"
        className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="demo-simple-select"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Health
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        disabled={false}
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value="Health"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-nnbavb"
        >
          <span
            className="notranslate"
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
